### PR TITLE
fix(issue-149): update email validation on input change

### LINKED ISSUE #149: 🐞 [Bug Report] - Click purchase without email disables purchase button until page refresh
**Describe the bug**

The purchase button in the cart page is disabled if I click the button without a valid email address. It is not re-enabled through regular navigation, only through refreshing the page.

**Expected behavior**

The purchase button resets once there is a state change to the email field.

**Screenshots**
Screenshot 2023-04-07 at 9 18 55 AM

### PR DESCRIPTION
**Pull request by pattishin**
## Description

Fixes #149 

To verify:

* Follow steps to run locally https://github.com/GoogleCloudPlatform/avocano/tree/main/client
* Add a product to your cart
* Attempt to check out and ensure that:
  * purchase button is disabled if no email input 
  * purchase button is disabled when email is not correct format
  * purchase button is enabled when email is in correct format

## Checklist
- [x] Added steps to reproduce the changes in this pull request
- [ ] Added relevant testing in this pull request
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: client/src/components/checkout-form.js [hunk 1]
```diff
@@ -29,22 +29,33 @@ class CheckoutForm extends LitElement {
...
+    const isValid = this.isValidEmail(form.get('email'));
+
+    // Disable submit while form is being sent
+    this.state.disableSubmit = !isValid;
     this.requestUpdate();
   }
 
   isValidEmail(text) {
-    return /[^ @]*@[^ @]*/.test(text);
+    return /^([\w\.\-]+)@([\w\-]+)((\.(\w){2,3})+)$/.test(text);
```
Comment (grayside) [inline client/src/components/checkout-form.js:58]: suggestion: This regex doesn't validate the full range of email addresses, and while that doesn't matter much for this app, folks may copy this to a production app.

user+label@example.com (+label denied).

**Comment by glasnt** [comment]
Preview errors won't be resolved until #240 merged, and this PR rebased to suit.

File: client/src/components/checkout-form.js (hunk 1 above)
Comment (glasnt) [reply client/src/components/checkout-form.js:58]: Agreed, would suggest reverting this change

```suggestion
    return /[^ @]*@[^ @]*/.test(text);
```

**Comment by glasnt** [comment]
Deployment preview: https://avocano-preview.web.app/

**Comment by pattishin** [comment]
Making a few updates! Will ping when ready for review again

**Review by grayside — APPROVED** [review]
LGTM based on code review.

File: client/package.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
     "test:watch": "web-test-runner --watch",
     "build": "rimraf dist && rollup -c rollup.config.js",
     "start:build": "web-dev-server --root-dir dist --app-index index.html --open",
-    "start": "web-dev-server"
+    "start": "web-dev-server",
+    "prepare": "cd .. && husky install client/.husky"
```
Comment (glasnt) [inline client/package.json:15]: Wondering what husky is; I think I've seen it block some commits. If this command should be mentioned in the client dev docs, please add it in a separate PR

**Pull request merged**